### PR TITLE
feat: add support to validate credentials for configured universe

### PR DESCRIPTION
**Pull request by ohmayr**
Towards b/324211188 🦕

### CODE REVIEW

File: googleapiclient/discovery.py [hunk 1]
```diff
@@ -53,6 +53,13 @@
 except ImportError:  # pragma: NO COVER
     google_auth_httplib2 = None
 
+try:
+    from google.api_core import universe as universe
```
Comment (parthea) [inline googleapiclient/discovery.py:57]:
```suggestion
    from google.api_core import universe
```

File: googleapiclient/discovery.py [hunk 2]
```diff
@@ -1352,6 +1359,7 @@ def __init__(
         resourceDesc,
         rootDesc,
         schema,
+        universe_domain=universe.DEFAULT_UNIVERSE if HAS_UNIVERSE else None,
```
Comment (parthea) [inline googleapiclient/discovery.py:1362]: Add type hints. Should the default value be `""`?

File: tests/test_discovery.py [hunk 3]
```diff
@@ -54,6 +54,13 @@
 except ImportError:
     HAS_OAUTH2CLIENT = False
 
+try:
+    from google.api_core import universe as universe
```
Comment (parthea) [inline tests/test_discovery.py:58]:
```suggestion
    from google.api_core import universe
```

File: tests/test_discovery.py [hunk 4]
```diff
@@ -2331,5 +2340,129 @@ def test_get_media(self):
...
+
+            # TODO(omairn): uncomment when universe configured universe is leveraged.
+            # http = google_auth_httplib2.AuthorizedHttp(credentials=None, http=build_http())
+            # discovery = read_datafile("zoo.json")
+            # service = build_from_document(discovery, http=http, client_options=google.api_core.client_options.ClientOptions(universe_domain=fake_universe))
+
+            # with self.assertRaises(universe.UniverseMismatchError):
+            #     service._validate_credentials()
+
+            # TODO(omairn): Add test case for not specifying universe domain via client option.
```
Comment (parthea) [inline tests/test_discovery.py:2371]: Please remove commented code and add TODO comments linking to a Github issue in this repo

**Review by parthea — APPROVED** [review]

File: tests/test_discovery.py [hunk 5]
```diff
@@ -2331,5 +2340,129 @@ def test_get_media(self):
...
+                discovery,
+                http=http,
+                client_options=google.api_core.client_options.ClientOptions(
+                    universe_domain=universe.DEFAULT_UNIVERSE
+                ),
+            )
+
+            assert service._validate_credentials()
+            assert service._credentials_validated
+            assert service._validate_credentials()
```
Comment (parthea) [inline tests/test_discovery.py:2441]:
```suggestion
            assert service._validate_credentials()
            assert service._credentials_validated
            assert service._validate_credentials()
```

Did you intend to have assert twice for assert service._validate_credentials()?

Comment (parthea) [reply tests/test_discovery.py:2441]: I see that this is intended based on the name of the test. Can you add a comment above the 2nd call?

File: googleapiclient/discovery.py (hunk 1 above)
Comment (ohmayr) [reply googleapiclient/discovery.py]: addressed.

File: tests/test_discovery.py (hunk 3 above)
Comment (ohmayr) [reply tests/test_discovery.py]: addressed.

File: tests/test_discovery.py (hunk 4 above)
Comment (ohmayr) [reply tests/test_discovery.py]: addressed.

File: tests/test_discovery.py (hunk 5 above)
Comment (ohmayr) [reply tests/test_discovery.py]: addressed.

**Review by parthea — APPROVED** [review]

**Pull request merged**